tests/test_kmp.py

```diff
@@ -0,0 +1,29 @@
+"""Module to test the KMP implementations."""
+
+from unittest import TestCase
+from algorithms.searching.kmp import KMP
+
+
+class TestKMP(TestCase):
```

Comment:
Can you try adding the test cases using the `parameterized` library like it is being used [here](https://github.com/albexl/data-structures-for-teaching/blob/dev/tests/test_sorting.py#L19)? Comment if you have any questions on how to do it.

Reply:
i'm going to look at the library